context.go

```diff
@@ -76,9 +76,7 @@ func getTChannelParams(ctx context.Context) *tchannelCtxParams {
 
 // NewContext returns a new root context used to make TChannel requests.
 func NewContext(timeout time.Duration) (context.Context, context.CancelFunc) {
-	return NewContextBuilder(timeout).
-		setSpan(NewRootSpan()).
```

Comment:
Setting parent (root) span on the context would indeed allow access to the trace ID prior to child span being created inside of tchannel.

The way `CurrentSpan(ctx)` works is it tries to find an OpenTracing Span in the context, and then tries to convert it to internal `Span` format (assuming tracer supports that). Otherwise it returns nil.